endpoint/endpoint.go

```diff
@@ -396,3 +397,41 @@ func RemoveDuplicates(endpoints []*Endpoint) []*Endpoint {
 
 	return result
 }
+
+// Check endpoint if is it properly formatted according to RFC standards
+func CheckEndpoint(endpoint Endpoint) bool {
+	switch recordType := endpoint.RecordType; recordType {
+	case "MX":
+		for _, target := range endpoint.Targets {
+			// MX records must have a preference value to indicate priority, e.g. "10 example.com"
+			// as per https://www.rfc-editor.org/rfc/rfc974.txt
+			targetParts := strings.Fields(strings.TrimSpace(target))
+			if len(targetParts) != 2 {
+				return false
+			}
+
+			preferenceRaw := targetParts[0]
+			_, err := strconv.ParseInt(preferenceRaw, 10, 32)
+			if err != nil {
+				return false
+			}
+		}
+	case "SRV":
+		for _, target := range endpoint.Targets {
```

Comment:
An optional, nice to have solution, is to move actual validation from this block to a Target

```
func (t Targets) ValidateSRVRecord() {
	...body...
}
func (t Targets) ValidateMXRecord() {
	...body...
}
```

Isolating the validation logic into separate functions makes it easier to update or modify the validation rules for each record type without affecting the main function. And should simplify testing as well